ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-against-materialized-aggregates.json

```diff
@@ -1927,17 +1914,103 @@
       }
     },
     {
-      "name": "IN: fail on non-literal key",
+      "name": "IN: fail on column reference key",
       "statements": [
         "CREATE STREAM INPUT (ID INT KEY, IGNORED INT) WITH (kafka_topic='test_topic', value_format='JSON');",
         "CREATE TABLE AGGREGATE AS SELECT ID, COUNT(1) AS COUNT FROM INPUT GROUP BY ID;",
-        "SELECT * FROM AGGREGATE WHERE ID IN (CAST(1 AS INTEGER));"
+        "SELECT * FROM AGGREGATE WHERE ID IN (COUNT + 1);"
       ],
       "expectedError": {
         "type": "io.confluent.ksql.rest.entity.KsqlStatementErrorMessage",
-        "message": "Only comparison to literals is currently supported: (ID IN (CAST(1 AS INTEGER)))",
+        "message": "Unsupported column reference in pull query: (COUNT + 1)",
```

Comment:
What should be supported is functions on literals, maps and arrays of literals but not functions on columns?

Reply:
yes, essentially anything static (column references can't be used anywhere)

Reply:
Discussed offline. In this PR we only want to support non-literal keys (maps and structs) and expressions that use only literals.